benches/search_comparison.rs

```diff
@@ -285,3 +257,18 @@ fn make_sorted_vec<T: Ord>(mut v: Vec<T>) -> Vec<T> {
 fn search_sorted_vec<'a, T: Ord>(c: &'a Vec<T>, x: T) -> Option<&'a T> {
     c.binary_search(&x).ok().map(|i| &c[i])
 }
+
+fn pseudorandom_iter<T>(max: usize) -> impl Iterator<Item = T>
+where
+    T: TryFrom<usize>,
+    <T as TryFrom<usize>>::Error: core::fmt::Debug,
+{
+    let mut seed = 0usize;
```

Comment:
This just made me realize that the construction benchmarks (and also the search benchmarks) are always benchmarking the _same_ set of numbers. I can't quite decide whether that's a feature (eliminate noise) or a bug (only benchmarking one particular data set).

Reply:
It is indeed a good question. I think it is beneficial to save PRNG state between benchmarking iterations. This way the sequence will be random on the level of different iteration but deterministic on the level of whole benchmark.

Reply:
could you check the code?